### PR TITLE
[Distributed] Generate SIL for DistributedActor.resolve

### PR DESCRIPTION
**Pull request by drexin**
rdar://78484431

### CODE REVIEW

**Comment by drexin** [comment]
@swift-ci smoke test

**Comment by drexin** [comment]
@swift-ci smoke test

File: lib/SILGen/SILGenDistributed.cpp [hunk 1]
```diff
@@ -494,21 +609,62 @@ void SILGenFunction::emitDistributedActorFactory(FuncDecl *fd) {
...
+      F.mapTypeIntoContext(selfTyDecl->getDeclaredInterfaceType()));
+
+  // ==== Prepare all the basic blocks
+  auto returnBB = createBasicBlock();
+  auto resolvedBB = createBasicBlock();
+  auto makeProxyBB = createBasicBlock();
+  auto switchBB = createBasicBlock();
+  auto errorBB = createBasicBlock();
+
+  SILFunctionConventions fnConv = F.getConventions(); // TODO: no idea?
```
Comment (ktoso) [inline lib/SILGen/SILGenDistributed.cpp:629]:
```suggestion
  SILFunctionConventions fnConv = F.getConventions(); 
```

seems to be right I guess :)

File: lib/SILGen/SILGenDistributed.cpp [hunk 2]
```diff
@@ -494,21 +609,62 @@ void SILGenFunction::emitDistributedActorFactory(FuncDecl *fd) {
...
+
+  // --- get the uninitialized allocation from the runtime system.
+  FullExpr scope(Cleanups, CleanupLocation(fd));
+
+  auto optionalReturnTy = SILType::getOptionalType(returnTy);
```
Comment (ktoso) [inline lib/SILGen/SILGenDistributed.cpp:634]: nice optionals made things a bit easier :)

File: lib/SILGen/SILGenDistributed.cpp [hunk 3]
```diff
@@ -494,21 +609,62 @@ void SILGenFunction::emitDistributedActorFactory(FuncDecl *fd) {
...
+  // ==== Case 'some') return the resolved instance
+  {
+    B.emitBlock(resolvedBB);
 
-    // --- get the uninitialized allocation from the runtime system.
-    FullExpr scope(Cleanups, CleanupLocation(fd));
+    auto local = resolvedBB->createPhiArgument(returnTy, OwnershipKind::Owned);
+
+    B.createBranch(loc, returnBB, {local});
+  }
```
Comment (ktoso) [inline lib/SILGen/SILGenDistributed.cpp:662]: oh nice, not too hard then 👍

File: lib/SILGen/SILGenDistributed.cpp [hunk 4]
```diff
@@ -563,9 +743,8 @@ void SILGenFunction::emitDistributedActor_resignAddress(
                              getLoweredType(idVarDeclRef->getType()));
 
   // ==== locate: self.actorTransport
-  auto transportVarDeclRefs = cd->lookupDirect(ctx.Id_actorTransport);
-  assert(transportVarDeclRefs.size() == 1);
-  auto *transportVarDeclRef = dyn_cast<VarDecl>(transportVarDeclRefs.front());
+  auto transportVarDeclRef = lookupActorTransportProperty(ctx, cd, selfValue);
+
```
Comment (ktoso) [inline lib/SILGen/SILGenDistributed.cpp:747]: 👍

File: lib/SILGen/SILGenFunction.cpp [hunk 5]
```diff
@@ -514,17 +514,18 @@ void SILGenFunction::emitFunction(FuncDecl *fd) {
...
   emitProlog(captureInfo, fd->getParameters(), fd->getImplicitSelfDecl(), fd,
              fd->getResultInterfaceType(), fd->hasThrows(), fd->getThrowsLoc());
-  prepareEpilog(true, fd->hasThrows(), CleanupLocation(fd));
 
   if (fd->isDistributedActorFactory()) {
     // Synthesize the factory function body
     emitDistributedActorFactory(fd);
   } else {
+    prepareEpilog(true, fd->hasThrows(), CleanupLocation(fd));
+
```
Comment (ktoso) [inline lib/SILGen/SILGenFunction.cpp:523]: I see, so we emit the cleanups ourselves ok 👍

File: stdlib/public/Distributed/ActorTransport.swift [hunk 6]
```diff
@@ -48,7 +48,7 @@ public protocol ActorTransport: Sendable {
   ///
   /// Detecting liveness of such remote actors shall be offered / by transport libraries
   /// by other means, such as "watching an actor for termination" or similar.
-  func resolve<Act>(_ identity: AnyActorIdentity, as actorType: Act.Type) throws -> ActorResolved<Act> // TODO(distributed): make just optional
+  func resolve<Act>(_ identity: AnyActorIdentity, as actorType: Act.Type) throws -> Act? // TODO(distributed): make just optional
```
Comment (ktoso) [inline stdlib/public/Distributed/ActorTransport.swift:51]:
```suggestion
  func resolve<Act>(_ identity: AnyActorIdentity, as actorType: Act.Type) throws -> Act?
```

**Review by ktoso — APPROVED** [review]
Looks good! We can remove the TODOs that this addressed I guess

**Comment by drexin** [comment]
@swift-ci smoke test

**Comment by drexin** [comment]
@swift-ci smoke test

**Comment by drexin** [comment]
@swift-ci smoke test

File: lib/IRGen/GenDistributed.cpp [hunk 7]
```diff
@@ -41,9 +47,11 @@ llvm::Value *irgen::emitDistributedActorInitializeRemote(
   call->setCallingConv(IGF.IGM.SwiftCC);
   call->setDoesNotThrow();
 
-  out.add(call);
+  auto result = IGF.Builder.CreateBitCast(call, destType);
```
Comment (ktoso) [inline lib/IRGen/GenDistributed.cpp:50]: huh I see, very good to know..! Thanks a lot!

**Pull request merged**